lib/fetchers/Permit2Fetcher.ts

```diff
@@ -7,16 +7,20 @@ import { log } from '../util/log';
 import { metrics } from '../util/metrics';
 
 export class Permit2Fetcher {
-  public readonly permit2Address: string;
   public readonly permit2Abi: ethers.ContractInterface;
   private readonly permit2: ethers.Contract;
+  private readonly permit2ZkSync: ethers.Contract;
   private readonly chainIdRpcMap: Map<ChainId, providers.StaticJsonRpcProvider>;
 
   constructor(chainIdRpcMap: Map<ChainId, providers.StaticJsonRpcProvider>) {
     this.chainIdRpcMap = chainIdRpcMap;
-    this.permit2Address = PERMIT2_ADDRESS;
     this.permit2Abi = PERMIT2_CONTRACT.abi;
-    this.permit2 = new ethers.Contract(this.permit2Address, this.permit2Abi);
+    this.permit2 = new ethers.Contract(permit2Address(), this.permit2Abi);
+    this.permit2ZkSync = new ethers.Contract(permit2Address(ChainId.ZKSYNC), this.permit2Abi);
```

Comment:
why are both needed? cant the constructor just check chainId and set `this.permit2` to the correct address for this chain?

Reply:
I think we can turn it into a chainIdPermit2 map, just like chainIdRpcMap